components/renku_data_services/project/db.py

```diff
@@ -80,41 +88,53 @@ async def insert_project(self, user: base_models.APIUser, project: models.Projec
             async with session.begin():
                 session.add(project_orm)
 
+                # TODO: Test that raising an exception causes the transaction to be aborted or not
```

Comment:
You can write a test about this if you want or test manually. But it would be nice to do this before we merge.

Reply:
This was a comment for myself to test this by hand. Basically, I wanted to make sure that this block works as a transaction. I believe it's not easy (and also needed) to write an automatic test for it.

Reply:
I'll remove all these TODOs in the final code.